src/openms/include/OpenMS/FILTERING/TRANSFORMERS/SpectraMerger.h

```diff
@@ -278,6 +284,240 @@ namespace OpenMS
+               }
+              
+              // go backward
+              steps = 0;
+              m = n;
+              for (typename MapType::const_iterator it_rt_2 = it_rt; (it_rt_2 != exp.begin() && (steps <= range_scans)); --it_rt_2)
+              {
+                if (Int(it_rt_2->getMSLevel()) == *it_mslevel)
+                {
+                  std::pair<Size, double> p(m,1);
+                  if (m!=n)    // already covered in forward case
+                  {
+                    spectra_to_average_over[n].push_back(p);
+                  }
+                  ++steps;
+                }
+                --m;
+              }
+            }
+            else    // RT unit = seconds
+            {
+              // go forward
+              m = n;
+              for (typename MapType::const_iterator it_rt_2 = it_rt; (it_rt_2 != exp.end() && (std::abs(it_rt_2->getRT() - it_rt->getRT()) <= range_seconds)); ++it_rt_2)
+              {
+                if (Int(it_rt_2->getMSLevel()) == *it_mslevel)
+                {                
+                  std::pair<Size, double> p(m,1);
+                  spectra_to_average_over[n].push_back(p);
+                }
+                ++m;
+               }
+              
+              // go backward
+              m = n;
+              for (typename MapType::const_iterator it_rt_2 = it_rt; (it_rt_2 != exp.begin() && (std::abs(it_rt_2->getRT() - it_rt->getRT()) <= range_seconds)); --it_rt_2)
+              {
+                if (Int(it_rt_2->getMSLevel()) == *it_mslevel)
+                {
+                  if (m!=n)    // already covered in forward case
+                  {
+                    std::pair<Size, double> p(m,1);
+                    spectra_to_average_over[n].push_back(p);
+                  }
+                }
+                --m;
+              }
+            }
+
+          }
+          ++n;
+        }
+
+        // normalize weights
+        for (AverageBlocks::Iterator it = spectra_to_average_over.begin(); it != spectra_to_average_over.end(); ++it)
+        {
+          double sum(0.0);
+          for (std::vector<std::pair<Size, double> >::const_iterator it2 = it->second.begin(); it2 != it->second.end(); ++it2)
+          {
+            sum += it2->second;
+          }
+          
+          for (std::vector<std::pair<Size, double> >::iterator it2 = it->second.begin(); it2 != it->second.end(); ++it2)
+          {
+            (*it2).second /= sum;
+          }
+        }
+        
+        // determine type of spectral data (profile or centroided)
+        Size idx = spectra_to_average_over.begin()->first;    // index of first spectrum to be averaged
+        SpectrumSettings::SpectrumType spectrum_type = exp[idx].getType();
+        if (spectrum_type == SpectrumSettings::UNKNOWN)
+        {
+          spectrum_type = PeakTypeEstimator().estimateType(exp[idx].begin(), exp[idx].end());
+        }
+        
+        if (spectrum_type == SpectrumSettings::PEAKS)
```

Comment:
well I dont think it is necessary for the user to be able to override the peakdetector only for certain MS levels. I think it would be sufficient to have automatic peak detection for most cases and an override mechanism that would allow to specify that the whole file is profile or centroid. For more complicated cases (e.g. swath files with centroid MS2 but profile MS1 or something equivalent...), the user then should use the FileFilter, apply the tool independently per ms level and potentially combine the results again